### PR TITLE
Merge 12 ➡️ 13

### PR DESCRIPTION
**Pull request by caguero**
# ➡️ Forward port

Port `gz-transport12` to `gz-transport13`

Branch comparison: https://github.com/gazebosim/gz-transport/compare/gz-transport13...gz-transport12

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

**Review by mjcarroll — APPROVED** [review]

File: src/cmd/gz_TEST.cc [hunk 1]
```diff
@@ -506,110 +369,100 @@ TEST(gzTest, TopicEcho)
...
-  std::filesystem::path scriptPath = PROJECT_SOURCE_DIR;
-  scriptPath = scriptPath / "src" / "cmd" / "transport.bash_completion.sh";
-
-  // Equivalent to:
-  // sh -c "bash -c \". /path/to/transport.bash_completion.sh;
-  // _gz_topic_flags\""
-  std::string cmd = "bash -c \". " + scriptPath.string() +
-    "; _gz_topic_flags\"";
-  std::string scriptOutput = custom_exec_str(cmd);
+// TEST(gzTest, TopicHelpVsCompletionFlags)
```
Comment (caguero) [inline src/cmd/gz_TEST.cc:436]: I think we need to restore these two tests, they were temporary commented by me but the intent was to port them.

File: src/cmd/gz_TEST.cc [hunk 2]
```diff
@@ -406,68 +400,68 @@ TEST(gzTest, TopicEchoNum)
...
+//   // Call the output function in the bash completion script
+//   std::filesystem::path scriptPath = PROJECT_SOURCE_DIR;
+//   scriptPath = scriptPath / "src" / "cmd" / "transport.bash_completion.sh";
+
+//   // Equivalent to:
+//   // sh -c "bash -c \". /path/to/transport.bash_completion.sh;
+//   // _gz_service_flags\""
+//   std::string cmd = "bash -c \". " + scriptPath.string() +
+//     "; _gz_service_flags\"";
+//   std::string scriptOutput = custom_exec_str(cmd);
```
Comment (scpeters) [inline src/cmd/gz_TEST.cc:417]: porting this will be a bit tricky, since the `custom_exec_str` method now prefixes all commands with `gz`

Comment (mjcarroll) [reply src/cmd/gz_TEST.cc:417]: I think we can just use the unwrapped `gz::utils::Subprocess` for this

File: src/cmd/gz_TEST.cc (hunk 1 above)
Comment (mjcarroll) [reply src/cmd/gz_TEST.cc:436]: #481

File: src/cmd/gz_TEST.cc (hunk 2 above)
Comment (mjcarroll) [reply src/cmd/gz_TEST.cc:417]: #481

**Pull request merged**